library/src/main/java/io/ona/kujaku/location/clients/BaseLocationClient.java

```diff
@@ -33,15 +36,44 @@ public LocationClientListener getListener() {
         return locationClientListener;
     }
 
+    /**
+     * @deprecated use {@code addLocationListener} instead
+     */
+    @Deprecated
     @Override
     public void setLocationListener(@Nullable LocationListener locationListener) {
-        this.locationListener = locationListener;
+        locationListeners.add(0, locationListener);
     }
 
+    /**
+     * @deprecated use {@code getLocationListeners} instead
+     */
+    @Deprecated
     @Nullable
     @Override
     public LocationListener getLocationListener() {
-        return locationListener;
+        if (locationListeners.size() > 0) {
+            return locationListeners.get(0);
+        }
+        return null;
+    }
+
+    @Override
+    public void addLocationListener(@NonNull LocationListener locationListener) {
```

Comment:
This method should have a boolean return type, that indicates whether the listener was added. This follows the same convention as `locationListeners.add` which returns a boolean. If the list already contains the listener then return false. 

What do you think?